### PR TITLE
ROS#110 - Cyclic tasklets

### PR DESCRIPTION
**Pull request by Glamhoth**

### CODE REVIEW

File: arch/x86/aerugo-x86-hal/src/hal.rs [hunk 1]
```diff
@@ -32,14 +31,16 @@ impl AerugoHal for Hal {
...
             TIME_START
                 .elapsed()
                 .expect("{}")
                 .as_nanos()
                 .try_into()
                 .unwrap(),
-        )
+        );
+
+        Instant::from_ticks(duration.ticks())
```
Comment (SteelPh0enix) [inline arch/x86/aerugo-x86-hal/src/hal.rs:43]: why like that, readability, or is there some difference in code underneath?

File: src/cyclic_execution.rs [hunk 2]
```diff
@@ -5,25 +5,48 @@
...
+            next_execution_time: Instant::from_ticks(0).into(),
+            period,
+            tasklet,
+        }
     }
 
-    /// Wakes that stored tasklet.
-    pub(crate) fn wake_tasklet(&self) {
-        Aerugo::wake_tasklet(&self.tasklet);
+    /// Wakes that stored tasklet if the time has come.
```
Comment (SteelPh0enix) [inline src/cyclic_execution.rs:31]:
```suggestion
    /// Wakes that stored tasklet if the time for it's execution has come.
```

"if the time has come" sounds pretty ominous

File: src/cyclic_execution.rs [hunk 3]
```diff
@@ -5,25 +5,48 @@
...
+    ///
+    /// # Parameters
+    /// * `current_time` - Current system time.
+    pub(crate) fn wake_if_time(&self, current_time: Instant) {
```
Comment (SteelPh0enix) [inline src/cyclic_execution.rs:35]: maybe `wake_if_should_execute`?

File: arch/x86/aerugo-x86-hal/src/hal.rs (hunk 1 above)
Comment (Glamhoth) [reply arch/x86/aerugo-x86-hal/src/hal.rs:43]: It was wrong before, it passed nanoseconds instead of ticks

**Review by SteelPh0enix — APPROVED** [review]

**Pull request merged**